lib/vagrant-openshift/action/run_openshift3_tests.rb

```diff
@@ -55,26 +55,38 @@ def run_tests(env, cmds, as_root=true)
           exit_code
         end
 
+        #
+        # Build and run the make commands
+        #   for testing all run make test
+        #   for testing unit tests only run make build check
+        #   for testing assets run hack/test-assets.sh
+        #
+        # All env vars will be added to the beginning of the command like VAR=1 make test
+        #
         def call(env)
           @options.delete :logs
 
-          if @options[:report_coverage]
-            cmds = ['OUTPUT_COVERAGE=/tmp/origin/e2e/artifacts/coverage hack/test-go.sh']
-          else
-            cmds = ['hack/test-go.sh']
-          end
+          cmd_env = []
+          build_targets = ["make"]
 
           if @options[:all]
-            cmds << 'hack/test-integration.sh'
-            cmds << 'hack/test-cmd.sh'
+            cmd_env << 'ARTIFACT_DIR=/tmp/origin/e2e/artifacts'
+            cmd_env << 'LOG_DIR=/tmp/origin/e2e/logs'
+            build_targets << 'test'
```

Comment:
Updated the make file to add a SKIP_BUILD flag that controls if the build is done or not.  https://github.com/openshift/origin/pull/886